packages/firestore/lib/index.d.ts

```diff
@@ -49,6 +49,39 @@ import { ReactNativeFirebase } from '@react-native-firebase/app';
  */
 export namespace FirebaseFirestoreTypes {
   import FirebaseModule = ReactNativeFirebase.FirebaseModule;
+  /**
+   * An instance of Filter used to generate Firestore Filter queries.
+   */
+  export interface Filter {
+    fieldPath: keyof T | FieldPath;
+    operator: WhereFilterOp;
+    value: any;
+  }
+  /**
+   * The Filter functions used to generate an instance of Filter.
+   */
+  export interface FilterFunction {
+    /**
+     * The Filter function used to generate an instance of Filter.
+     * e.g. Filter('name', '==', 'Ada')
+     */
+    (fieldPath: keyof T | FieldPath, operator: WhereFilterOp, value: any): Filter;
+    /**
+     * The Filter.or() static function used to generate a logical OR query using multiple Filter instances.
+     * e.g. Filter.or(Filter('name', '==', 'Ada'), Filter('name', '==', 'Bob'))
+     */
+    or(...queries: Filter[]): Filter;
+    /**
+     * The Filter.and() static function used to generate a logical AND query using multiple Filter instances.
+     * e.g. Filter.and(Filter('name', '==', 'Ada'), Filter('name', '==', 'Bob'))
+     */
+    and(...queries: Filter[]): Filter;
```

Comment:
@russellwheatley 

🤔  these types don't seem like a fit vs the firebase-js-sdk types, is that intentional?

Here's the `and()` from firebase-js-sdk https://firebase.google.com/docs/reference/js/firestore_.md#and

Has all sorts of QueryFilterCompositeConstraint stuff, takes QueryFilterConstraint[] etc types that aren't present here

Not sure how this will work as a drop-in replacement

Reply:
I've updated types to match naming: https://github.com/invertase/react-native-firebase/pull/7045/commits/20e9c8d7f200a9b00c461b54d9659b3afb2b6305. Let me know if this works.